src/libs/elektra/keyhelpers.c

```diff
@@ -195,13 +195,9 @@ int keyNameIsSystem (const char * name)
  * clears key (all data members are set to zero)
  * Initializes an empty metadata keyset if null or clears it.
  */
-int keyInit (Key * key)
+void keyInit (Key * key)
 {
-	if (key->meta) ksClear (key->meta);
```

Comment:
If it is possible that someone calls `keyInit` with a key including metadata, maybe we should `ksDel(key->meta)`.
Otherwise maybe just update the documentation.

Reply:
As far as I know this is not the case anywhere in `elektra-core`, where `keyInit` is only called on key creation or similar. Not sure about bindings.

If we have to `ksDel(key->meta)`, then we again have to set `key->meta = NULL` before all  `keyInit` calls, otherwise we pass a garbage pointer to `ksDel`.

Reply:
> otherwise we pass a garbage pointer to ksDel.

Not if they are allocated using calloc.

Reply:
`keyInit` is not listed in the relevant `symbols.map` and therefore is not accessible outside of `elektra-core`. Inside of `elektra-core` it is only called on uninitialised keys. This is explains the modifications to `keyInit` I made in #3115. I'm not sure you can do the same without some of the other changes of #3115, but in any case it doesn't really matter how you solve it, since #3115 will change it anyway.